### PR TITLE
[Feature] Support backup and restore without broker

### LINKED ISSUE #9320: Support backup and restore without broker

### PR DESCRIPTION
**Pull request by xiaoyong-z**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #9320

## Problem Summary(Required) ：

Support backup and restore without broker

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 34 / 310 (10.97%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/task/UploadTask.java|0|16|00.00%|[51, 52, 53, 54, 55, 56, 57, 77, 78, 80, 83, 84, 85, 87, 88, 89]|
|:large_blue_circle:|com/starrocks/backup/RestoreJob.java|0|18|00.00%|[1030, 1031, 1032, 1033, 1034, 1035, 1037, 1038, 1039, 1041, 1042, 1043, 1044, 1045, 1121, 1122, 1125, 1126]|
|:large_blue_circle:|com/starrocks/task/DownloadTask.java|0|16|00.00%|[50, 51, 52, 53, 54, 55, 56, 76, 77, 79, 82, 83, 84, 86, 87, 88]|
|:large_blue_circle:|com/starrocks/fs/HdfsUtil.java|0|65|00.00%|[73, 74, 75, 76, 81, 136, 138, 139, 140, 141, 143, 148, 150, 151, 152, 153, 154, 155, 167, 178, 180, 181, 182, 183, 184, 185, 200, 276, 277, 278, 279, 280, 284, 285, 286, 287, 288, 289, 291, 292, 295, 296, 297, 298, 299, 300, 303, 307, 308, 312, 314, 316, 317, 318, 323, 324, 327, 328, 329, 330, 332, 333, 343, 344, 378]|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsService.java|0|1|00.00%|[63]|
|:large_blue_circle:|com/starrocks/backup/BlobStorage.java|17|161|10.56%|[113, 116, 133, 134, 308, 309, 311, 313, 314, 315, 318, 319, 321, 322, 323, 324, 325, 326, 330, 332, 333, 335, 336, 337, 338, 341, 342, 343, 344, 345, 346, 347, 350, 351, 352, 355, 356, 357, 358, 359, 360, 365, 366, 368, 371, 372, 373, 374, 375, 376, 378, 381, 382, 383, 388, 389, 438, 440, 441, 442, 443, 444, 448, 449, 577, 579, 581, 582, 583, 585, 586, 587, 588, 590, 592, 594, 595, 596, 599, 600, 601, 604, 605, 606, 607, 608, 609, 611, 612, 613, 614, 615, 616, 619, 622, 623, 624, 626, 630, 631, 676, 679, 680, 681, 682, 683, 685, 686, 687, 691, 692, 735, 736, 737, 738, 739, 740, 741, 747, 748, 798, 799, 800, 801, 802, 803, 804, 805, 806, 807, 808, 810, 814, 815, 832, 833, 881, 882, 883, 885, 886, 887, 889, 1021]|
|:large_blue_circle:|com/starrocks/analysis/CreateRepositoryStmt.java|3|7|42.86%|[53, 94, 95, 96]|
|:large_blue_circle:|com/starrocks/backup/BackupJob.java|9|19|47.37%|[504, 505, 509, 511, 512, 513, 514, 515, 535, 536]|
|:large_blue_circle:|com/starrocks/backup/BackupHandler.java|3|5|60.00%|[182, 183]|
|:large_blue_circle:|com/starrocks/backup/Status.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 34 / 311 (10.93%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/task/UploadTask.java|0|16|00.00%|[51, 52, 53, 54, 55, 56, 57, 77, 78, 80, 83, 84, 85, 87, 88, 89]|
|:large_blue_circle:|com/starrocks/backup/RestoreJob.java|0|18|00.00%|[1030, 1031, 1032, 1033, 1034, 1035, 1037, 1038, 1039, 1041, 1042, 1043, 1044, 1045, 1121, 1122, 1125, 1126]|
|:large_blue_circle:|com/starrocks/task/DownloadTask.java|0|16|00.00%|[50, 51, 52, 53, 54, 55, 56, 76, 77, 79, 82, 83, 84, 86, 87, 88]|
|:large_blue_circle:|com/starrocks/fs/HdfsUtil.java|0|65|00.00%|[73, 74, 75, 76, 81, 136, 138, 139, 140, 141, 143, 148, 150, 151, 152, 153, 154, 155, 167, 178, 180, 181, 182, 183, 184, 185, 200, 276, 277, 278, 279, 280, 284, 285, 286, 287, 288, 289, 291, 292, 295, 296, 297, 298, 299, 300, 303, 307, 308, 312, 314, 316, 317, 318, 323, 324, 327, 328, 329, 330, 332, 333, 343, 344, 378]|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsService.java|0|1|00.00%|[63]|
|:large_blue_circle:|com/starrocks/backup/BlobStorage.java|17|162|10.49%|[113, 116, 133, 134, 200, 308, 309, 311, 313, 314, 315, 318, 319, 321, 322, 323, 324, 325, 326, 330, 332, 333, 335, 336, 337, 338, 341, 342, 343, 344, 345, 346, 347, 350, 351, 352, 355, 356, 357, 358, 359, 360, 365, 366, 368, 371, 372, 373, 374, 375, 376, 378, 381, 382, 383, 388, 389, 438, 440, 441, 442, 443, 444, 448, 449, 577, 579, 581, 582, 583, 585, 586, 587, 588, 590, 592, 594, 595, 596, 599, 600, 601, 604, 605, 606, 607, 608, 609, 611, 612, 613, 614, 615, 616, 619, 622, 623, 624, 626, 630, 631, 676, 679, 680, 681, 682, 683, 685, 686, 687, 691, 692, 735, 736, 737, 738, 739, 740, 741, 747, 748, 798, 799, 800, 801, 802, 803, 804, 805, 806, 807, 808, 810, 814, 815, 832, 833, 881, 882, 883, 885, 886, 887, 889, 1021]|
|:large_blue_circle:|com/starrocks/analysis/CreateRepositoryStmt.java|3|7|42.86%|[53, 94, 95, 96]|
|:large_blue_circle:|com/starrocks/backup/BackupJob.java|9|19|47.37%|[504, 505, 509, 511, 512, 513, 514, 515, 535, 536]|
|:large_blue_circle:|com/starrocks/backup/BackupHandler.java|3|5|60.00%|[182, 183]|
|:large_blue_circle:|com/starrocks/backup/Status.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 34 / 311 (10.93%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/task/UploadTask.java|0|16|00.00%|[51, 52, 53, 54, 55, 56, 57, 77, 78, 80, 83, 84, 85, 87, 88, 89]|
|:large_blue_circle:|com/starrocks/backup/RestoreJob.java|0|18|00.00%|[1030, 1031, 1032, 1033, 1034, 1035, 1037, 1038, 1039, 1041, 1042, 1043, 1044, 1045, 1121, 1122, 1125, 1126]|
|:large_blue_circle:|com/starrocks/task/DownloadTask.java|0|16|00.00%|[50, 51, 52, 53, 54, 55, 56, 76, 77, 79, 82, 83, 84, 86, 87, 88]|
|:large_blue_circle:|com/starrocks/fs/HdfsUtil.java|0|65|00.00%|[73, 74, 75, 76, 81, 136, 138, 139, 140, 141, 143, 148, 150, 151, 152, 153, 154, 155, 167, 178, 180, 181, 182, 183, 184, 185, 200, 276, 277, 278, 279, 280, 284, 285, 286, 287, 288, 289, 291, 292, 295, 296, 297, 298, 299, 300, 303, 307, 308, 312, 314, 316, 317, 318, 323, 324, 327, 328, 329, 330, 332, 333, 343, 344, 378]|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsService.java|0|1|00.00%|[63]|
|:large_blue_circle:|com/starrocks/backup/BlobStorage.java|17|162|10.49%|[113, 116, 133, 134, 200, 308, 309, 311, 313, 314, 315, 318, 319, 321, 322, 323, 324, 325, 326, 330, 332, 333, 335, 336, 337, 338, 341, 342, 343, 344, 345, 346, 347, 350, 351, 352, 355, 356, 357, 358, 359, 360, 365, 366, 368, 371, 372, 373, 374, 375, 376, 378, 381, 382, 383, 388, 389, 438, 440, 441, 442, 443, 444, 448, 449, 577, 579, 581, 582, 583, 585, 586, 587, 588, 590, 592, 594, 595, 596, 599, 600, 601, 604, 605, 606, 607, 608, 609, 611, 612, 613, 614, 615, 616, 619, 622, 623, 624, 626, 630, 631, 676, 679, 680, 681, 682, 683, 685, 686, 687, 691, 692, 735, 736, 737, 738, 739, 740, 741, 747, 748, 798, 799, 800, 801, 802, 803, 804, 805, 806, 807, 808, 810, 814, 815, 832, 833, 881, 882, 883, 885, 886, 887, 889, 1021]|
|:large_blue_circle:|com/starrocks/analysis/CreateRepositoryStmt.java|3|7|42.86%|[53, 94, 95, 96]|
|:large_blue_circle:|com/starrocks/backup/BackupJob.java|9|19|47.37%|[504, 505, 509, 511, 512, 513, 514, 515, 535, 536]|
|:large_blue_circle:|com/starrocks/backup/BackupHandler.java|3|5|60.00%|[182, 183]|
|:large_blue_circle:|com/starrocks/backup/Status.java|2|2|100.00%|[]|

**Review by chaoyli — DISMISSED** [review]

**Comment by xiaoyong-z** [comment]
run starrocks_admit_test

**Comment by xiaoyong-z** [comment]
run starrocks_admit_test

**Comment by xiaoyong-z** [comment]
run starrocks_admit_test

File: be/src/fs/fs_s3.cpp [hunk 1]
```diff
@@ -303,6 +305,35 @@ StatusOr<std::unique_ptr<WritableFile>> S3FileSystem::new_writable_file(const Wr
     return std::make_unique<OutputStreamAdapter>(std::move(ostream), fname);
 }
 
+Status S3FileSystem::rename_file(const std::string& src, const std::string& target) {
+    S3URI src_uri;
+    S3URI dest_uri;
+    if (src_uri.parse(src)) {
```
Comment (meegoo) [inline be/src/fs/fs_s3.cpp:311]:
```suggestion
    if (!src_uri.parse(src)) {
```

File: fe/fe-core/src/main/java/com/starrocks/backup/BlobStorage.java [hunk 2]
```diff
@@ -276,8 +304,90 @@ public Status downloadWithFileSize(String remoteFilePath, String localFilePath,
...
+                                    + " file size: %d, file: %s. msg: %s",
+                            readOffset, readLen, fileSize,
+                            remoteFilePath, e.getMessage());
+                    LOG.warn(lastErrMsg);
+                    status = new Status(ErrCode.COMMON_ERROR, lastErrMsg);
+                    break;
+                }
+
+                // NOTICE(cmy): Sometimes the actual read length does not equal to the expected read length,
+                // even if the broker's read buffer size is large enough.
```
Comment (meegoo) [inline fe/fe-core/src/main/java/com/starrocks/backup/BlobStorage.java:363]: misleading comments

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 34 / 308 (11.04%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/task/UploadTask.java|0|16|00.00%|[51, 52, 53, 54, 55, 56, 57, 77, 78, 80, 83, 84, 85, 87, 88, 89]|
|:large_blue_circle:|com/starrocks/backup/RestoreJob.java|0|18|00.00%|[1030, 1031, 1032, 1033, 1034, 1035, 1037, 1038, 1039, 1041, 1042, 1043, 1044, 1045, 1121, 1122, 1125, 1126]|
|:large_blue_circle:|com/starrocks/task/DownloadTask.java|0|16|00.00%|[50, 51, 52, 53, 54, 55, 56, 76, 77, 79, 82, 83, 84, 86, 87, 88]|
|:large_blue_circle:|com/starrocks/fs/HdfsUtil.java|0|65|00.00%|[73, 74, 75, 76, 81, 136, 138, 139, 140, 141, 143, 148, 150, 151, 152, 153, 154, 155, 167, 178, 180, 181, 182, 183, 184, 185, 200, 276, 277, 278, 279, 280, 284, 285, 286, 287, 288, 289, 291, 292, 295, 296, 297, 298, 299, 300, 303, 307, 308, 312, 314, 316, 317, 318, 323, 324, 327, 328, 329, 330, 332, 333, 343, 344, 378]|
|:large_blue_circle:|com/starrocks/fs/hdfs/HdfsService.java|0|1|00.00%|[63]|
|:large_blue_circle:|com/starrocks/backup/BlobStorage.java|17|159|10.69%|[113, 116, 133, 134, 200, 308, 309, 311, 313, 314, 315, 318, 319, 321, 322, 323, 324, 325, 326, 330, 332, 333, 335, 336, 337, 338, 341, 342, 343, 344, 345, 346, 347, 350, 351, 352, 355, 356, 357, 358, 359, 360, 365, 366, 367, 368, 369, 370, 372, 375, 376, 377, 382, 383, 432, 434, 435, 436, 437, 438, 442, 443, 571, 573, 575, 576, 577, 579, 580, 581, 582, 584, 586, 588, 589, 590, 593, 594, 595, 598, 599, 600, 601, 602, 603, 605, 606, 607, 608, 609, 610, 613, 616, 617, 618, 620, 624, 625, 670, 673, 674, 675, 676, 677, 679, 680, 681, 685, 686, 729, 730, 731, 732, 733, 734, 735, 741, 742, 792, 793, 794, 795, 796, 797, 798, 799, 800, 801, 802, 804, 808, 809, 826, 827, 875, 876, 877, 879, 880, 881, 883, 1015]|
|:large_blue_circle:|com/starrocks/analysis/CreateRepositoryStmt.java|3|7|42.86%|[53, 94, 95, 96]|
|:large_blue_circle:|com/starrocks/backup/BackupJob.java|9|19|47.37%|[504, 505, 509, 511, 512, 513, 514, 515, 535, 536]|
|:large_blue_circle:|com/starrocks/backup/BackupHandler.java|3|5|60.00%|[182, 183]|
|:large_blue_circle:|com/starrocks/backup/Status.java|2|2|100.00%|[]|

**Comment by xiaoyong-z** [comment]
run starrocks_be_unittest

**Comment by xiaoyong-z** [comment]
run starrocks_be_unittest

**Comment by xiaoyong-z** [comment]
run starrocks_admit_test

**Comment by xiaoyong-z** [comment]
run starrocks_be_unittest

**Review by meegoo — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**